pkg/entrypoint/server_test.go

```diff
@@ -0,0 +1,197 @@
+package entrypoint
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+)
+
+const apiAddr = "localhost:8080"
+
+type testCase struct {
+	input    []Job
+	expected []statusResponse
+}
+type statusResponse struct {
+	Jobs []job
+}
+type job struct {
+	Name   string
+	Status string
+}
+
+var log logr.Logger
+var _ = BeforeSuite(func() {
+	zapLog, _ := zap.NewDevelopment()
+	log = zapr.NewLogger(zapLog)
+})
+
+var _ = Describe("entrypoint status API test", func() {
+	It("should state of successful command changed from running to completed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "sleep",
+					Args:    []string{"1"},
+				},
+			},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Running"}}},
+				{Jobs: []job{{Name: "test1", Status: "Completed"}}},
```

Comment:
I believe that checking the progress along the way will cause unintended test failures if the timing is off.
It would be better to separate the test cases that check for Runninng during execution by using commands that do not complete, such as `sleep inf` and `pause`, and those that check for Completed at the end of execution.

Reply:
@kmdkuk 
Are you concerned that the test code cannot observe `Running` status and fails? If so, I agree with you.

However, it is required to check that the status is first `Running` and then changed to `Completed`. I think that it is enough to extend sleep duration, say 5s.

Reply:
@umezawatakeshi 
> Are you concerned that the test code cannot observe Running status and fails? If so, I agree with you.

Yes.

> However, it is required to check that the status is first Running and then changed to Completed. I think that it is enough to extend sleep duration, say 5s.

I thought this could be fulfilled by separating the test cases and checking them separately, but is that not good enough?

Reply:
@kmdkuk 

> I thought this could be fulfilled by separating the test cases and checking them separately, but is that not good enough?

No. Separated tests cannot check transition.

Reply:
@yamatcha I'm sorry, I have asked you to fix the test to check the progress, but it looks like that part should be put back the way it was.

Reply:
OK
I implement a test to check that the status first becomes Running and then changes to Completed.